### PR TITLE
Use the new get out of date connectors endpoint

### LINKED ISSUE #20256: Use new Out of Date Connectors Endpoint on the Frontend
Implement the work done in https://github.com/airbytehq/airbyte/issues/19853 on the frontend.

### PR DESCRIPTION
**Pull request by krishnaglick**
## What
Resolves https://github.com/airbytehq/airbyte/issues/20256 and https://github.com/airbytehq/airbyte/issues/19081

## How
Uses the new get out of date connectors code endpoint instead of loading all definitions and latest, and determining it on the front end.
Some minor improvements made to the upgrade connectors view.

### CODE REVIEW

File: airbyte-webapp/src/services/connector/ConnectorDefinitions.ts [hunk 1]
```diff
@@ -36,3 +41,22 @@ export const useConnectorSpecifications = (): ConnectorSpecifications => {
...
+
+  const requestAuthMiddleware = useDefaultRequestMiddlewares();
+
+  return useInitService(() => new ConnectorService(apiUrl, requestAuthMiddleware), [apiUrl, requestAuthMiddleware]);
+}
+
+export const useGetOutOfDateConnectorsCount = () => {
+  const service = useConnectorService();
+  return useSuspenseQuery(connectorDefinitionKeys.count, () => service.checkUpdates());
+};
```
Comment (krishnaglick) [inline airbyte-webapp/src/services/connector/ConnectorDefinitions.ts:62]: TODO: Probably split this out some.

File: airbyte-webapp/src/views/layout/SideBar/NotificationIndicator.tsx [hunk 2]
```diff
@@ -12,7 +12,8 @@ const Notification = styled(Indicator)`
 `;
 
 export const NotificationIndicator: React.FC = () => {
-  const { hasNewVersions } = useConnector();
+  const { hasNewVersions, outOfDateConnectors } = useGetConnectorsOutOfDate();
+  console.log(outOfDateConnectors);
```
Comment (krishnaglick) [inline airbyte-webapp/src/views/layout/SideBar/NotificationIndicator.tsx:16]: TODO: Remove

File: airbyte-webapp/src/hooks/services/useConnector.tsx [hunk 3]
```diff
@@ -51,6 +34,18 @@ const useConnector = (): ConnectorService => {
...
+
+export const useUpdateDestinationDefinitions = () => {
+  const { destinationDefinitions } = useDestinationDefinitionList();
+  const { mutateAsync: updateDestinationDefinition } = useUpdateDestinationDefinition();
+
+  const newDestinationDefinitions = useMemo(
+    () => destinationDefinitions.filter(Connector.hasNewerVersion),
+    [destinationDefinitions]
+  );
+
```
Comment (krishnaglick) [inline airbyte-webapp/src/hooks/services/useConnector.tsx:48]: These two methods were just moved into their own hooks so definitions aren't loaded for the count any more.

File: airbyte-webapp/src/hooks/services/useConnector.test.tsx [hunk 4]
```diff
@@ -1,73 +0,0 @@
-import { act, renderHook } from "@testing-library/react-hooks";
```
Comment (krishnaglick) [inline airbyte-webapp/src/hooks/services/useConnector.test.tsx:1]: The testing could likely be rewritten, but the method signatures are really light comparatively.

File: airbyte-webapp/src/pages/SettingsPage/pages/ConnectorsPage/components/ConnectorsView.tsx [hunk 5]
```diff
@@ -99,13 +99,14 @@ const ConnectorsView: React.FC<ConnectorsViewProps> = ({
                   onChange={onUpdateVersion}
                   feedback={feedbackList[Connector.id(row.original)]}
                   currentVersion={row.original.dockerImageTag}
+                  updating={loading}
```
Comment (krishnaglick) [inline airbyte-webapp/src/pages/SettingsPage/pages/ConnectorsPage/components/ConnectorsView.tsx:102]: Added this so you can't try to upgrade a connector after hitting upgrade all.

File: airbyte-webapp/src/services/connector/DestinationDefinitionService.ts [hunk 6]
```diff
@@ -117,6 +118,8 @@ const useUpdateDestinationDefinition = () => {
             ) ?? [],
         })
       );
+
+      queryClient.invalidateQueries(connectorDefinitionKeys.count());
```
Comment (krishnaglick) [inline airbyte-webapp/src/services/connector/DestinationDefinitionService.ts:122]: This seems to be the best way to handle this. Open to suggestions.

**Comment by krishnaglick** [comment]
@timroes This code exists across team lines. Should I put Lake on this PR?

**Comment by krishnaglick** [comment]
Screenshot 2022-12-16 at 2 06 23 PM
Screenshot 2022-12-16 at 2 06 30 PM

Lighthouse score for the Connection list page. 6 point improvement! Sources and destination list pages still make the old call so they don't get as much of a boost.

**Review by flash1293 — APPROVED** [review]
Tested and works as expected, would prefer a second look from @lmossman though.

Doesn't need to be part of this PR, but it seems like in most places we are any loading both the workspace list _and_ the latest list - would it make sense to collapse this information into a single request on the server so we don't need to fetch the list twice in any place?

**Comment by krishnaglick** [comment]
> Tested and works as expected, would prefer a second look from @lmossman though.
> 
> Doesn't need to be part of this PR, but it seems like in most places we are any loading both the workspace list _and_ the latest list - would it make sense to collapse this information into a single request on the server so we don't need to fetch the list twice in any place?

Work to do this is ticketed! :)

**Review by lmossman — APPROVED** [review]
Changes LGTM

**Pull request merged**